src/gui/dialogs/window_dlg_wait.c

```diff
@@ -68,32 +63,35 @@ void window_dlg_wait_draw(window_dlg_wait_t *window) {
         }
     }
 
-    if (window->flags & DLG_W8_HOUR_CHNG) { //hourglass animation (just sand 0x0100, whole icon 0x0200)
-        rect_ui16_t icon_rc = rc;
-        icon_rc.h = icon_rc.w = 30;
-        icon_rc.x += 110;
-        icon_rc.y += 50;
+    if (window->components & DLG_W8_DRAW_HOURGLASS) { // hourglass animation enabled
+        if (window->animation_chng) {                 //hourglass animation timeout
+            window->animation_chng = false;
+
+            rect_ui16_t icon_rc = rc;
+            icon_rc.h = icon_rc.w = 30;
+            icon_rc.x += 110;
+            icon_rc.y += 50;
+            render_icon_align(icon_rc, IDR_PNG_wizard_icon_hourglass, COLOR_BLACK, ALIGN_CENTER);
 
-        if (window->flags & DLG_W8_HOUR_REDRW) {
             const uint16_t x = icon_rc.x;
             const uint16_t y = icon_rc.y;
 
-            render_icon_align(icon_rc, IDR_PNG_wizard_icon_hourglass, COLOR_BLACK, ALIGN_CENTER);
-            if (window->flags & 0x0010) {
+            // if (window->animation == ANIM_START) -> draw only icon
+            if (window->animation == ANIM_SAND_1) {
```

Comment:
how about a switch(){...}?

Reply:
And the sequences of lines can be also reduced by using some arrays of coordinates to draw. Not a blocker though, just says something about the author's programming abilities ;)

Reply:
This hardcoded coordinates are so ugly it is not in my powers to make it better. I tried, but it's probably less readable..
commit `0ace17e`